File: src/main/java/net/dumbdogdiner/dogcore/commands/mute/UnmuteCommand.java
package net.dumbdogdiner.dogcore.commands.mute;

import dev.jorel.commandapi.annotations.Command;
import dev.jorel.commandapi.annotations.Default;
import dev.jorel.commandapi.annotations.Permission;
import dev.jorel.commandapi.annotations.arguments.AOfflinePlayerArgument;
import net.dumbdogdiner.dogcore.Permissions;
import net.dumbdogdiner.dogcore.database.User;
import net.dumbdogdiner.dogcore.messages.Messages;
import org.bukkit.OfflinePlayer;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

@Command("unmute")
@Permission(Permissions.MUTE)
public final class UnmuteCommand {
    private UnmuteCommand() { }

    @Default
    public static void unmute(
        final @NotNull CommandSender sender,
        final @NotNull @AOfflinePlayerArgument OfflinePlayer player
    ) {
        User.lookupCommand(player, sender, user -> user.unmute().thenAccept(v ->
            sender.sendMessage(Messages.get("commands.unmute.success", user.formattedName().join()))));
    }
}

File: src/main/kotlin/net/dumbdogdiner/dogcore/commands/UnmuteCommand.kt
package net.dumbdogdiner.dogcore.commands

import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.offlinePlayerArgument
import net.dumbdogdiner.dogcore.db.DbPlayer
import net.kyori.adventure.text.Component
import org.bukkit.OfflinePlayer

fun unmuteCommand() = commandAPICommand("unmute") {
    withPermission("dogcore.unmute")

    offlinePlayerArgument("player")

    anyExecutor { sender, args ->
        val player = DbPlayer.lookup(args["player"] as OfflinePlayer)
        if (player == null) {
            sender.sendMessage(Component.text("Player has no records in the server."))
            return@anyExecutor
        }
        player.unmute()
    }
}
